receiver/snmpreceiver/scraper_test.go

```diff
@@ -0,0 +1,2409 @@
+					expectedMetrics, err := golden.ReadMetrics(goldenPath)
+					require.NoError(t, err)
+					return expectedMetrics
+				}
+				expectedMetrics := expectedMetricGen(t)
+				metrics, err := scraper.scrape(context.Background())
+				require.NoError(t, err)
+				err = scrapertest.CompareMetrics(expectedMetrics, metrics)
+				require.NoError(t, err)
+			},
+		},
+		{
+			desc: "Simple scalar scrape with non '.' prefixed OID still creates metric (1)",
+			testFunc: func(t *testing.T) {
+				mockClient := new(MockClient)
+				clientSNMPData := snmpData{
+					oid:       ".1",
+					value:     int64(1),
+					valueType: integerVal,
+				}
+				mockClient.On("Connect").Return(nil)
+				mockClient.On("Close").Return(nil)
+				mockClient.On("GetScalarData", mock.Anything, mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
+					processFn := args.Get(1).(processFunc)
+					returnErr := processFn(clientSNMPData)
+					require.NoError(t, returnErr)
+				}).Return(nil)
+				scraper := &snmpScraper{
+					cfg: &Config{
+						Metrics: map[string]*MetricConfig{
+							"metric1": {
+								Description: "test description",
+								Unit:        "By",
+								Gauge: &GaugeMetric{
+									ValueType: "int",
+								},
+								ScalarOIDs: []ScalarOID{
+									{
+										OID: "1",
+									},
+								},
+							},
+						},
+					},
+					settings: componenttest.NewNopReceiverCreateSettings(),
+					client:   mockClient,
+					logger:   zap.NewNop(),
+				}
+
+				expectedMetricGen := func(t *testing.T) pmetric.Metrics {
+					goldenPath := filepath.Join("testdata", "expected_metrics", "1_scalar_metrics_golden.json")
+					expectedMetrics, err := golden.ReadMetrics(goldenPath)
+					require.NoError(t, err)
+					return expectedMetrics
+				}
+				expectedMetrics := expectedMetricGen(t)
+				metrics, err := scraper.scrape(context.Background())
+				require.NoError(t, err)
+				err = scrapertest.CompareMetrics(expectedMetrics, metrics)
+				require.NoError(t, err)
+			},
+		},
+		{
+			desc: "Simple scalar scrape creates float cumulative monotonic sum metric (2)",
+			testFunc: func(t *testing.T) {
+				mockClient := new(MockClient)
+				clientSNMPData := snmpData{
+					oid:       ".1",
+					value:     float64(1.0),
+					valueType: floatVal,
+				}
+				mockClient.On("Connect").Return(nil)
+				mockClient.On("Close").Return(nil)
+				mockClient.On("GetScalarData", mock.Anything, mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
+					processFn := args.Get(1).(processFunc)
+					returnErr := processFn(clientSNMPData)
+					require.NoError(t, returnErr)
+				}).Return(nil)
+				scraper := &snmpScraper{
+					cfg: &Config{
+						Metrics: map[string]*MetricConfig{
+							"metric1": {
+								Description: "test description",
+								Unit:        "By",
+								Sum: &SumMetric{
+									Aggregation: "cumulative",
+									Monotonic:   true,
+									ValueType:   "float",
```

Comment:
I noticed that `mdatagen` uses `double` instead of "float". We should change this to be consistent.

Reply:
@djaglowski i'll make these changes even though i don't think keeping consistent with "mdatagen" is completely necessary. (one is used by devs, the other is used by people configuring the receiver)

Reply:
I agree it's not strictly necessary but the metadata side of the metric data model may yet be formalized in a shared struct. Might as well keep things as close as possible by default.